src/client/components/ReferenceSearchDialog/ReferenceSearchDialog.react.js

```diff
@@ -338,10 +345,14 @@ export default class ReferenceSearchDialog extends Component {
                           id={column.name}
                           type="text"
                           className="form-control"
-                          onChange={(event) => this.onChangeSearchParam(
-                            column.name, event.target.value
+                          onChange={(event) => this.onChangeSearchParams(
+                            { [column.name]: event.target.value }
                           )}
                           value={searchParams[column.name] || ''}
+                          {...(additionalProps && additionalProps({
+                            searchParams,
+                            onChangeSearchParams: this.onChangeSearchParams
```

Comment:
Not clear why 'searchParams' and 'onChangeSearchParams' are passed into onChangeSearchParams?